$OpenBSD: patch-src_connections_c,v 1.23 2013/03/28 16:22:32 brad Exp $

Call ERR_clear_error only for SSL connections in CON_STATE_ERROR.

--- src/connections.c.orig	Wed Mar 27 09:17:03 2013
+++ src/connections.c	Wed Mar 27 09:17:37 2013
@@ -1736,8 +1736,8 @@ int connection_state_machine(server *srv, connection *
 						break;
 					}
 				}
+				ERR_clear_error();
 			}
-			ERR_clear_error();
 #endif
 
 			switch(con->mode) {
